spark/src/test/scala/org/apache/spark/sql/delta/DeltaTableFeatureSuite.scala

```diff
@@ -256,6 +256,25 @@ class DeltaTableFeatureSuite
         .canUpgradeTo(Protocol(1, 1)))
   }
 
+  test("protocol downgrade compatibility") {
+    val tableFeatureProtocol =
+      Protocol(TABLE_FEATURES_MIN_READER_VERSION, TABLE_FEATURES_MIN_WRITER_VERSION)
+    // Cannot downgrade when at a minimum writer features are not supported.
+    assert(!Protocol(1, 6).canDowngradeTo(Protocol(1, 6)))
+    // Protocol version downgrades are not supported.
+    assert(!Protocol(3, 7).withFeature(TestWriterFeature).canDowngradeTo(Protocol(2, 7)))
```

Comment:
Let's also try downgrading to Protocol(1, 7). Reader protocol version 2 implicitly contains Column Mapping, which means that this downgrade may be rejected because moving to reader protocol 2 is technically an _upgrade_. Downgrading from read protocol version 3 with no features to read protocol version 1 is actually a no-op from a functionality perspective, so that would _truly_ test if we reject protocol version downgrades purely because they're protocol version downgrades.

And let's test separately that we can't downgrade the _writer_ protocol as well -- from 7 to 1 and from 7 to some value in [2..6].